airflow/models/dagrun.py

```diff
@@ -398,7 +398,9 @@ def generate_run_id(run_type: DagRunType, execution_date: datetime) -> str:
 
     @provide_session
     def get_task_instances(
-        self, state: Optional[Iterable[TaskInstanceState]] = None, session=None
+        self,
+        state: Optional[Iterable[Optional[TaskInstanceState]]] = None,
```

Comment:
Isn't `None` a valid TaskInstanceState? https://github.com/apache/airflow/blob/820e836c4a2e45239279d4d71e1db9434022fec5/airflow/utils/state.py#L35

Reply:
I suspect subclassing from `str` is the problem here (but we probably can’t fix that without breaking a ton of things).